MobileCenter/MobileCenter/Internals/Channel/MSLogManagerDelegate.h

```diff
@@ -1,46 +1,75 @@
 
 /**
  * Callback that is called when a log has been persisted successfully. This was introduced to implement the
  * log buffer for Crashes.
  *
- *  @param log The log.
- *  @param internalId An internal Id that can be used to keep track of logs.
+ * @param log The log.
+ * @param internalId An internal Id that can be used to keep track of logs.
  *
- *  @discussion We had some discussion about the naming of the method. To match the
- *  onEnqueueingLog:withInternalId (@see onEnqueueingLog:withInternalId) callback, it should
- *  also use `enqueuing` in it's signature, yet, as of now, it indicates successful persistence of a log. This
- *  method's name might change in the future to make a distinction between a successfully enqueued log and a persisted
- *  log.
+ * @discussion We had some discussion about the naming of the method. To match the
+ * onEnqueueingLog:withInternalId (@see onEnqueueingLog:withInternalId) callback, it should
+ * also use `enqueuing` in it's signature, yet, as of now, it indicates successful persistence of a log. This
+ * method's name might change in the future to make a distinction between a successfully enqueued log and a persisted
+ * log.
  */
 - (void)onFinishedPersistingLog:(id<MSLog>)log withInternalId:(NSString *)internalId;
 
 /**
- *  Callback that is called when persisting a log has failed, meaning it has not been saved to disk because the log was
- *  empty. This was introduced to implement the log buffer for Crashes.
+ * Callback that is called when persisting a log has failed, meaning it has not been saved to disk because the log was
+ * empty. This was introduced to implement the log buffer for Crashes.
  *
- *  @param log The log.
- *  @param internalId An internal Id that can be used to keep track of logs.
+ * @param log The log.
+ * @param internalId An internal Id that can be used to keep track of logs.
  *
- *  @discussion We had some discussion about the naming of the method. To match the
- *  onEnqueueingLog:withInternalId (@see onEnqueueingLog:withInternalId) callback, it should
- *  also use `enqueuing` in it's signature, yet, as of now, it indicates successful persistence of a log. This
- *  method's name might change in the future to make a distinction between a successfully enqueued log and a persisted
- *  log.
+ * @discussion We had some discussion about the naming of the method. To match the
+ * onEnqueueingLog:withInternalId (@see onEnqueueingLog:withInternalId) callback, it should
+ * also use `enqueuing` in it's signature, yet, as of now, it indicates successful persistence of a log. This
+ * method's name might change in the future to make a distinction between a successfully enqueued log and a persisted
+ * log.
  */
 - (void)onFailedPersistingLog:(id<MSLog>)log withInternalId:(NSString *)internalId;
 
+/**
+ * Callback method that will be called before each log will be send to the server.
+ *
+ * @param log The log to be sent.
+ */
+- (void)willSendLog:(id<MSLog>)log;
```

Comment:
Usually delegates should also reference the sender for instance `- (void)logManager:(id<MSLogManager>) willSendLog:(id<MSLog>)log;`. That's because this protocol doesn't know about who will implement it. Here we are making the assumption that only object already having a logManager object will implement this protocol and that may not always be true. (The other delegates before this one will have to be revised too be that's another story). @TroubleMakerBen and @jaelim-ms any thoughts?